src/hyp3_sdk/jobs.py

```diff
@@ -127,6 +127,7 @@ def download_files(self, location: Path | str = '.', create: bool = True) -> lis
         if not self.succeeded():
             raise HyP3SDKError(f'Only succeeded jobs can be downloaded; job is {self.status_code}.')
         if self.expired():
+            assert isinstance(self.expiration_time, datetime)
```

Comment:
I think this is more clear, since it makes it clear that the other possible value is `None`, rather than some other type of object:
```suggestion
            assert self.expiration_time is not None
```

Reply:
I don't agree with this. The point of `mypy` is enforcing that something is exactly the correct type. If we just trust that the object passed will be the correct type, then there is no point in type checking.

Reply:
But it does enforce that it's the correct type, because it's type-hinted where it's defined, right? We can discuss further.

Reply:
In `Job.__init__` it's defined as `expiration_time: datetime | None`, so `mypy` will enforce that it's one of those types. The `assert` is just to enforce that it's not `None` at that point.